tests/sqlparser_mysql.rs

```diff
@@ -2530,6 +2530,16 @@ fn parse_rlike_and_regexp() {
     }
 }
 
+#[test]
+fn parse_like_with_escape() {
+    mysql().verified_only_select(r#"SELECT * FROM customer WHERE name LIKE 'a\%c'"#);
```

Comment:
I took a look at the comments and the tests but Im not sure I currently understood the behavior/problem being solved for.

Figured here to try ask with a concrete example, given this test case with the string `'a\%c'` what was the previous vs current behavior was it the case that we failed to parse the string in the previous behavior or that it produced a different kind of output?

Reply:
Ah yeah that makes sense, thanks for clarifying!